client/src/pages/Tree/treeutilities.js

```diff
@@ -11,3 +11,19 @@ export async function updateOrCreateTree(id, fullData, updatedData) {
     await mutateTreeData.mutateAsync({ id, ...updatedData });
   }
 }
+
+const undefRequiredField = ((requiredField) => typeof requiredField === 'undefined');
+
+export const checkForMissingData = (currentTreeData) => [currentTreeData?.common,
+  currentTreeData?.scientific]
+  .some(undefRequiredField)
+    || ['vacant',
+      'vacant site',
+      'unsuitable site',
+      'asphalted well',
+      'planting site',
+      'other',
+      '#n/a', '::'].includes(currentTreeData?.common.toLowerCase());
```

Comment:
Given that this is only used in one component, I don't think it needs to be in its own file, and the updateOrCreateTree() function above this should be deleted anyway.

More importantly, having `vacant` or `asphalted well` in the `common` field doesn't mean the entry is missing data.  You should be able to add a note or edit the common name and check the "New tree" option to indicate a new tree as planted in that location.  You just can't maintain or change the health of a vacant site.

Reply:
> Issue is we don't have it set up to deal with the id change that will happen when common changes.
We will have to update the id or add a new tree with a different Id. Potential for multiple records with same lng lat.

That's an issue for any tree, then.  Sounds like the Edit link should be removed for all trees until the ID handling is addressed.